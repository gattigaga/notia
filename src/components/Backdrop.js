import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0px;
  left: 0px;
  background: rgba(222, 226, 226, 0.8);
  display: ${props => (props.isOpen ? "static" : "none")};
`;

const Backdrop = ({ isOpen, children, onClick }) => (
  <Container isOpen={isOpen} onClick={onClick}>
    {children}
  </Container>
);

Backdrop.propTypes = {
  isOpen: PropTypes.bool,
  children: PropTypes.node,
  onClick: PropTypes.func
};

export default Backdrop;
