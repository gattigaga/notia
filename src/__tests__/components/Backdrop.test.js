import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Backdrop from "components/Backdrop";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(
    <Backdrop onClick={onClick} isOpen>
      <h1>Hello World</h1>
    </Backdrop>
  );
});

describe("Backdrop", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls onClick", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
