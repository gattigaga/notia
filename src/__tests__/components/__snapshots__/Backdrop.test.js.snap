// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop renders correctly 1`] = `
<styled.div
  isOpen={true}
  onClick={[Function]}
>
  <h1>
    Hello World
  </h1>
</styled.div>
`;
